modal/client.py

```diff
@@ -352,26 +351,46 @@ async def _call_safely(self, coro, readable_method: str):
             logger.warning(f"RPC request to {readable_method} made outside of task context")
             return await coro
 
+    async def reset_on_pid_change(self):
+        if self._owner_pid and self._owner_pid != os.getpid():
+            await self._close()
+            self._stub = None
+            # TODO(elias): reset _cancellation_context?
+            await self._open()
+            try:
+                await self._init()
+            except BaseException:
+                await self._close()
+                raise
+
+    async def _get_grpclib_method(self, method_name: str) -> Any:
+        # safely get grcplib method that is bound to a valid channel
+        # This prevents usage of stale methods across forks of processes
+        await self.reset_on_pid_change()
+        return getattr(self._grpclib_stub, method_name)
+
     @synchronizer.nowrap
     async def _call_unary(
         self,
-        grpclib_method: grpclib.client.UnaryUnaryMethod[RequestType, ResponseType],
-        request: RequestType,
+        method_name: str,
+        request: Any,
```

Comment:
Why do we need to relax the typing so much here?

Reply:
Since we needed a method lookup layer of indirection at some point in the call chain in order to look up the "latest" grpclib method (since they are bound to channels) we need to break type safety at some point, but these private methods (_call_unary and _call_streaming) should only really be used at one place each within our code base, and those places (ie. `UnaryUnaryWrapper.__call__` etc) will expose proper types to the rest of the code base when used. Does this make sense?